### ISSUE TITLE
Staking-info command returns a ResourceExhausted error

### ISSUE THREAD

**Issue by vishalchangrani** [onflow/flow-cli#1277]
### Problem

flow accounts staking-info 0xbf91cf149c03a3b3 -n mainnet

❌ Command Error: error getting total stake for node: client: rpc error: code = ResourceExhausted desc = request is rate limited

### Acceptance Criteria

This command should return the staking info.

### Context

The error is incorrect. The rate limits are definitely high enough to accommodate to one request per second.
The command however works against a different access node - 
`$ flow accounts staking-info 0xbf91cf149c03a3b3  --host="access-001.mainnet24.nodes.onflow.org:9000"`

If a 50ms sleep is added here https://github.com/onflow/flow-cli/blob/master/internal/accounts/staking-info.go#L120 then the command works against the public access node (access.mainnet.nodes.onflow.org:9000) as well. This indicates that the flow cli times out prematurely and throws an incorrect error after that.

**Comment by bluesign** [comment]
I believe we should deal with the "request is rate limited" issue by making sure to retry a few times before giving up on all our calls ( by checking rate limit response info). I think the best spot to incorporate this is within the `flow-go-sdk` `Client`.

**Comment by peterargue** [comment]
Adding that 50ms sleep means that the loop will make at most 20 requests per second. I suspect that it's then sending fewer requests than the limit.

fwiw, that command appears to be working now, so the header changes QN made seems to have fixed this.

**Comment by bjartek** [comment]
I totally agree that the client should be smart and know about rate limits and headers and do the right thing.
